pkg/storage/postgres/postgres.go

```diff
@@ -0,0 +1,168 @@
+package postgres
+
+import (
+	"context"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+type Storage struct {
+	db *pgxpool.Pool
+}
+
+func New(constr string) (*Storage, error) {
+	db, err := pgxpool.Connect(context.Background(), constr)
+	if err != nil {
+		return nil, err
+	}
+	s := Storage{
+		db: db,
+	}
+	return &s, nil
+}
+
+type Task struct {
+	ID         int
+	Opened     int64
+	Closed     int64
+	AuthorID   int
+	AssignedID int
+	Title      string
+	Content    string
+}
+
+func (s *Storage) NewTask(t Task) (int, error) {
+	var id int
+	err := s.db.QueryRow(context.Background(), `
+		INSERT INTO tasks (author_id, assigned_id, title, content)
+		VALUES ($1, $2, $3, $4) RETURNING id;
+		`,
+		t.AuthorID,
+		t.AssignedID,
+		t.Title,
+		t.Content,
+	).Scan(&id)
+	return id, err
+}
+
+func (s *Storage) Tasks() ([]Task, error) {
+	rows, err := s.db.Query(context.Background(), `
+		SELECT
+			id,
+			opened,
+			closed,
+			author_id,
+			assigned_id,
+			title,
+			content
+		FROM tasks
+		ORDER BY id;
+	`)
+	return query(err, rows)
+}
+
+func query(err error, rows pgx.Rows) ([]Task, error) {
+	if err != nil {
+		return nil, err
+	}
+	var tasks []Task
+
+	for rows.Next() {
+		var t Task
+		err = rows.Scan(
+			&t.ID,
+			&t.Opened,
+			&t.Closed,
+			&t.AuthorID,
+			&t.AssignedID,
+			&t.Title,
+			&t.Content,
+		)
+		if err != nil {
+			return nil, err
+		}
+		tasks = append(tasks, t)
+	}
+	return tasks, rows.Err()
```

Comment:
больше кода, но выглядит лучше
```suggestion
        err = rows.Err()
        if err != nil {
            return nil, fmt.Errorf("......% %v", err)
        }
        
	return tasks, nil
```